projetos/axios-api-request/script.js

```diff
@@ -1,28 +1,157 @@
-	const ul = document.querySelector('ul');
+	const pokemonGridContainer = document.getElementById(
+		'pokemon-grid-container'
+	);
+	const pokemonGridItem = pokemonGridContainer.appendChild(
+		document.createElement('div')
+	);
+	pokemonGridItem.id = 'pokemon-grid-item';
 	//iterate over all pokemons
+	let count = 0;
 	pokemons.map(pokemon => {
-		ul.appendChild(createLi(pokemon));
-		console.log(pokemon);
+		count += 1;
+		pokemonGridItem.appendChild(createPokemonGridItem(pokemon, count));
 	});
 };
 
 const fetchPokemons = () => {
 	axios
-		.get('https://pokeapi.co/api/v2/pokemon/')
+		.get('https://pokeapi.co/api/v2/pokemon/?limit=151&offset=0')
 		.then(response => {
 			const pokemons = response.data['results'];
-			console.log(`GET list pokemons`, pokemons);
 			appendToDOM(pokemons);
 		})
 		.catch(error => console.error(error));
 };
 
 fetchPokemons();
+
+// filter
+function filter() {
+	var input, filter;
+	input = document.getElementById('pokemon-search');
+	filter = input.value.toLowerCase();
+	filterTest(filter);
+}
+function filterTest(filter) {
+	pokemonArray.map(pokemon => {
+		let pokemonDiv = document.getElementById(
+			`pokemon-name-image-${pokemon.name}`
+		);
+		if (filter === '') {
+			pokemonDiv.style.display = 'block';
+		} else if (pokemon.name.indexOf(filter) > -1) {
+			pokemonDiv.style.display = 'block';
+		} else {
+			pokemonDiv.style.display = 'none';
+		}
+	});
+}
+
+//add Pokémon
+function addPoke(newPoke) {
+	const pokemonGridItem = document.getElementById('pokemon-grid-item');
+	var newPoke = prompt(
+		'Great job on catching a new Pokémon! Type the new Pokémon Name'
+	).toLowerCase();
+	axios
+		.get(`https://pokeapi.co/api/v2/pokemon/${newPoke}`)
+		.then(response => {
+			newPokemonName = response.data['name'];
+			newPokemonId = response.data['id'];
+			pokemon = {
+				id: newPokemonId,
+				name: newPokemonName,
+				url: `https://pokeapi.co/api/v2/pokemon/${newPokemonId}/`
+			};
+			if (checkIfPokeIsOnList(pokemon) == false) {
+				pokemonGridItem.appendChild(
+					createPokemonGridItem(pokemon, newPokemonId)
+				);
+				alert(newPokemonName + ' added to list.');
+			} else if (checkIfPokeIsOnList(pokemon) == true) {
+				alert(newPokemonName + ' is already on the list.');
+			}
+		})
+		.catch(err => {
+			alert('Pokémon not found, please try again');
+			console.log('Error: ', err);
+		});
+}
+//remove Pokemon
+function removePoke(pokeToRemove) {
+	var pokeToRemove = prompt(
+		'Type the name of the Pokémon you want to remove.'
+	).toLowerCase();
+	const pokemonItem = document.getElementById(
+		`pokemon-name-image-${pokeToRemove}`
+	);
+	axios
+		.get(`https://pokeapi.co/api/v2/pokemon/${pokeToRemove}`)
+		.then(response => {
+			pokemonName = response.data['name'];
+			pokemonId = response.data['id'];
+			pokemon = {
+				id: pokemonId,
+				name: pokemonName,
+				url: `https://pokeapi.co/api/v2/pokemon/${pokemonId}/`
+			};
+			if (checkIfPokeIsOnList(pokemon) == true) {
```

Comment:
```suggestion
			if (checkIfPokeIsOnList(pokemon)) {
```

Quando você coloca dessa forma, ele compara se é true automaticamente =)